api/src/main/java/org/pmiops/workbench/db/model/DbAddress.java

```diff
@@ -52,7 +52,12 @@ public void setStreetAddress2(String streetAddress2) {
     this.streetAddress2 = streetAddress2;
   }
 
-  @Column(name = "zip_code")
+  // Most @Column annotations in our codebase don't have a length specification. This is included
+  // on the zip_code field to allow test cases (where an in-memory H2 database is used instead of
+  // MySQL) to trigger an exception when a user attempts to save a DbUser row with too-large field
+  // payloads. See ProfileControllerTest for the test case, and the Liquibase changelogs for the
+  // SQL definition of this field.
+  @Column(name = "zip_code", length = 10)
```

Comment:
US zip+4 is already 9 characters. I don't know if any international postal codes matter. Maybe pick a different field for this?

Also, we can define fake static entity classes in test classes or configs if this is really for test cases.

Also `EXPLAIN user` will give a good SQL definition in a table.

Reply:
10 is the actual SQL column length for zip_code – see comments in https://precisionmedicineinitiative.atlassian.net/browse/RW-5624 for my notes. This annotation won't change any production behavior, it just makes the in-memory database used for tests as stringent as SQL for this single field.

This was my compromise to get some sort of unit-level test replicating the behavior that we've seen in production. I don't love it, and spent quite some time trying to get a partial mock working, but eventually backed out and went with this approach.

Reply:
re: using or creating a test class that mocks UserService and uses doThrow... that's what I attempted to do by using a partial mock / spy within ProfileControllerTest, but (1) almost all existing controller tests rely on actual UserService behavior and (2) I couldn't get @SpyBean to work with our current version of Spring Boot (c.f. my notes in #workbrench-techdesign).

The right long-term solution here would be to test only controller-level logic in the controller test. That would probably require thinking hard about each test case, and bumping lots of tested logic down into UserServiceTest rather than ProfileControllerTest.

One short-term approach could have been to create a separate file for additional controller-level tests, but that complicates things by removing the strong pattern we have for one test file per class. Instead, I found a slightly hackier but simpler way of reproducing the errors related to the bug I picked up while oncall. 🤷

Reply:
I believe @jmthibault79 and I have already started moving some things downward into the service tests.

I'm generally less concerned by having additional test classes (especially if each one is simpler) than having to work with spies, but that's mainly because it's hard to reason about the internal state of a partially mocked object, especially if it's a complex class that gets a lot of edits. Having separate "themed" profile controller tests could help find the right places to break the controller down.